lib/chef/resource/launchd.rb

```diff
@@ -129,7 +129,7 @@ class Launchd < Chef::Resource
       property :abandon_process_group, [ TrueClass, FalseClass ],
         description: "If a job dies, all remaining processes with the same process ID may be kept running. Set to true to kill all remaining processes."
 
-      property :associated_bundle_identifiers, Hash,
+      property :associated_bundle_identifiers, Array,
```

Comment:
Seems like this needs to at a minimum be isolated to only macOS platforms. Would also like to see some testing around this to ensure that it's being handled correctly for macOS and non-macOS platforms